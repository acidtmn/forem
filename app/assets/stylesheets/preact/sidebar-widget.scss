@import 'variables';

.sidebar-additional {
  .widget {
    .widget-suggested-follows-container {
      h4 {
        margin: 0;
      }
      .widget-body {
        .widget-list-item__suggestions {
          display: flex;
          align-items: center;

          .widget-list-item__avatar {
            width: 48px;
            height: 48px;

            img {
              width: 100%;
              height: 100%;
              border-radius: 100%;
            }
          }

          .widget-list-item__content {
            flex: 1 auto;
            padding-left: 8px;
            h5 {
              all: unset;
              display: block;
              margin-bottom: 4px;
            }
          }

          .widget-list-item__follow-button {
            border: 2px solid transparent;
            border-radius: 3px;
            padding: 2px 8px;
            text-align: center;
            background-color: $green;
            color: $black;
            font-family: $helvetica-condensed;
            font-weight: 900;
            font-size: 0.8em;
          }
        }
      }
    }
  }
}
