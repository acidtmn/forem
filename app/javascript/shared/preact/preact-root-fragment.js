/**
 * A Preact 11+ implementation of the `replaceNode` parameter from Preact 10.
 *
 * This creates a "Persistent Fragment" (a fake DOM element) containing one or more
 * DOM nodes, which can then be passed as the `parent` argument to Preact's `render()` method.
 * Source: https://gist.github.com/developit/f4c67a2ede71dc2fab7f357f39cff28c
 */
export function createRootFragment(parent, replaceNode) {
  replaceNode = [].concat(replaceNode);
  var s = replaceNode[replaceNode.length-1].nextSibling;
  function insert(c, r) { parent.insertBefore(c, r || s); }
  return parent.__k = {
    nodeType: 1,
    parentNode: parent,
    firstChild: replaceNode[0],
    childNodes: replaceNode,
    insertBefore: insert,
    appendChild: insert,
    removeChild: function(c) { parent.removeChild(c); }
  };
}