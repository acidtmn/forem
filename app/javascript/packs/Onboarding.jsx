import { h, render } from 'preact';
import { getUserDataAndCsrfToken } from '../chat/util';
import getUnopenedChannels from '../src/utils/getUnopenedChannels';

HTMLDocument.prototype.ready = new Promise(resolve => {
  if (document.readyState !== 'loading') {
    return resolve();
  }
  document.addEventListener('DOMContentLoaded', () => resolve());
  return null;
});

function shouldShowOnboarding() {
  const { user } = document.body.dataset;

  return (
    document.head.getElementsByTagName('meta')[2].content === 'true' &&
    user &&
    JSON.parse(user).saw_onboarding === false
  );
}

function renderPage() {
  if (shouldShowOnboarding()) {
    import('../src/Onboarding').then(({ default: Onboarding }) =>
      render(<Onboarding />, document.getElementById('top-bar')),
    );
  }
}

document.ready.then(
  getUserDataAndCsrfToken().then(({ currentUser, csrfToken }) => {
    window.currentUser = currentUser;
    window.csrfToken = csrfToken;

    renderPage();
    getUnopenedChannels();
  }),
);
