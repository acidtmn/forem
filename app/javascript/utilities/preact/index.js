import { render as preactRender } from 'preact';
import { unmountComponentAtNode } from 'preact/compat';

/**
 * Renders a Preact component into the given parent DOM element.
 *
 * Note: This function is calling Preact's render under the hood along with
 * a call to InstantClick's change event that explicitly unmounts the component rendered.
 *
 * @param {ComponentChild} vnode
 * @param {Element | Document | ShadowRoot | DocumentFragment} parent
 * @param {Element | Text} replaceNode
 */
export function render(vnode, parent, replaceNode) {
  InstantClick.on('change', () => {
    parent && unmountComponentAtNode(parent);
  });

  preactRender(vnode, parent, replaceNode);
}
