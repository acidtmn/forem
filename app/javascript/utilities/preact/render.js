import { render as preactRender } from 'preact';
import { unmountComponentAtNode } from 'preact/compat';

/**
 * Renders a Preact component into the given parent DOM element.
 *
 * Note: This function is calling Preact's render under the hood along with
 * a call to InstantClick's change event that explicitly unmounts the component rendered.
 *
 * @param {object} vnode The virtual DOM node representing a component.
 * @param {Element | Document | ShadowRoot | DocumentFragment} parent The DOM element where the component will render.
 * @param {Element | Text} replaceNode An optional DOM element that must of the given parent.
 * Instead of inferring where to start rendering, it will update or replace the passed element using Preact's diffing algorithm.
 */
export function render(vnode, parent, replaceNode) {
  InstantClick.on('change', () => {
    // We need to explicitly unmount a Preact component since we are using InstantClick. Failing to do so
    // would leave references to past components.
    parent && unmountComponentAtNode(parent);
  });

  preactRender(vnode, parent, replaceNode);
}
