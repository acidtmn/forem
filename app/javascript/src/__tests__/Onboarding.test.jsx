import { h } from 'preact';
import { shallow, deep } from 'preact-render-spy';
import fetch from 'jest-fetch-mock';
import Onboarding from '../Onboarding';

global.fetch = fetch;

function flushPromises() {
  return new Promise(resolve => setImmediate(resolve));
}

describe('<Onboarding />', () => {
  afterEach(() => {
    document.body.setAttribute('data-user', null);
  });

  const fakeResponse = JSON.stringify([
    {
      bg_color_hex: '#000000',
      id: 715,
      name: 'discuss',
      text_color_hex: '#ffffff',
    },
    {
      bg_color_hex: '#f7df1e',
      id: 6,
      name: 'javascript',
      text_color_hex: '#000000',
    },
    {
      bg_color_hex: '#2a2566',
      id: 630,
      name: 'career',
      text_color_hex: '#ffffff',
    },
  ]);

  const dataUser = JSON.stringify({
    followed_tag_names: [
      {
        bg_color_hex: '#000000',
        id: 715,
        name: 'discuss',
        text_color_hex: '#ffffff',
      },
      {
        bg_color_hex: '#f7df1e',
        id: 6,
        name: 'javascript',
        text_color_hex: '#000000',
      },
    ],
  });

  const usersState = [
    {
      id: 1,
      name: 'Ben Halpern',
      profile_image_url: 'ben.jpg',
    },
    {
      id: 2,
      name: 'Krusty the Clown',
      profile_image_url: 'clown.jpg',
    },
    {
      id: 3,
      name: 'dev.to staff',
      profile_image_url: 'dev.jpg',
    },
  ];

  describe('when user is not logged in', () => {
    beforeEach(() => {
      fetch.mockResponse(fakeResponse);
      document.body.setAttribute(
        'data-user',
        JSON.stringify({ saw_onboarding: true }),
      );
    });

    test('showOnboarding is false', () => {
      const context = shallow(<Onboarding />);
      expect(context.state('showOnboarding')).toEqual(false);
    });

    test('nothing should be rendered', async () => {
      // TODO: dataUser should be {};
      const context = shallow(<Onboarding />);
      await flushPromises();
      expect(context).toMatchSnapshot();
    });
  });

  describe('Welcome page', () => {
    let context;
    beforeEach(async () => {
      fetch.mockResponse(fakeResponse);
      document.body.setAttribute('data-user', dataUser);
      context = deep(<Onboarding />);
      await flushPromises();
    });

    test('renders properly', () => {
      expect(context).toMatchSnapshot();
    });

    test('NEXT button should work', () => {
      context.find('.button').simulate('click');
      expect(context.state('pageNumber')).toEqual(2);
      expect(context).toMatchSnapshot();
    });

    // TODO: test closing button
  });

  describe('Follow tag page', () => {
    let context;
    beforeEach(async () => {
      fetch.mockResponseOnce(fakeResponse);
      document.body.setAttribute('data-user', dataUser);
      context = deep(<Onboarding />);
      context.setState({ pageNumber: 2 });
      await flushPromises();
    });

    test('renders properly', () => {
      expect(context).toMatchSnapshot();
    });

    test('BACK button works', () => {
      context
        .find('.button')
        .at(0)
        .simulate('click');
      expect(context.state('pageNumber')).toEqual(1);
      expect(context).toMatchSnapshot();
    });

    test('NEXT button works', async () => {
      fetch.mockResponseOnce(JSON.stringify(usersState));
      context
        .find('.button')
        .at(1)
        .simulate('click');
      await flushPromises();
      expect(context.state('pageNumber')).toEqual(3);
      expect(context).toMatchSnapshot();
    });
  });

  describe('Follow dev members page', () => {
    let context;
    beforeEach(async () => {
      fetch.mockResponse(fakeResponse);
      document.body.setAttribute('data-user', dataUser);
      context = shallow(<Onboarding />);
      context.setState({ pageNumber: 3, usersState });
      await flushPromises();
    });

    test('renders properly', () => {
      expect(context).toMatchSnapshot();
    });

    test('BACK button works', () => {
      context
        .find('.button')
        .at(0)
        .simulate('click');
      expect(context.state('pageNumber')).toEqual(2);
      expect(context).toMatchSnapshot();
    });

    test('NEXT button works', () => {
      context
        .find('.button')
        .at(1)
        .simulate('click');
      expect(context.state('pageNumber')).toEqual(4);
      expect(context).toMatchSnapshot();
    });
  });

  xit('display if there is current user', async () => {
    document.body.setAttribute('data-user', dataUser);
    fetch.mockResponse(fakeResponse);
    const context = shallow(<Onboarding />);
    await flushPromises();
    expect(context).toMatchSnapshot();
  });

  xit('allows user to interact properly', async () => {
    document.body.setAttribute('data-user', dataUser);
    const meta = document.createElement('meta');
    meta.setAttribute('name', 'csrf-token');
    document.body.appendChild(meta);
    fetch.mockResponseOnce(fakeResponse);

    const context = deep(<Onboarding />, { depth: 3 });
    // context.setState({ users: usersState });
    // context.rerender();
    await flushPromises();
    context.find('.button').simulate('click');

    // going to page two
    expect(context.state('pageNumber')).toEqual(2);
    expect(context.find('[onClick]')).toMatchSnapshot();

    // going to page three
    context
      .find('.button')
      .at(1)
      .simulate('click');
    console.log(context.state());
    expect(context.state('pageNumber')).toEqual(3);
    // going to page four
    context
      .find('.button')
      .at(1)
      .simulate('click');
    expect(context.state('pageNumber')).toEqual(4);
    // going to page five
    context
      .find('.button')
      .at(1)
      .simulate('click');
    expect(context.state('pageNumber')).toEqual(5);
    fetch.mockResponse(JSON.stringify({ outcome: 'onboarding closed' }));
    // going back to page four
    context
      .find('.button')
      .at(0)
      .simulate('click');
    expect(context.state('pageNumber')).toEqual(4);
    // going back to page five
    context
      .find('.button')
      .at(1)
      .simulate('click');
    expect(context.state('pageNumber')).toEqual(5);
    // evaluting the button text on page five
    expect(
      context
        .find('[onClick]')
        .at(2)
        .text(),
    ).toEqual("LET'S GO");
    context
      .find('.button')
      .at(1)
      .simulate('click');
    // clicking the final button
    await flushPromises();
    expect(context.state('showOnboarding')).toEqual(false);
  });

  xit('allow user to follow tags flawlessly', async () => {
    const dataUser = JSON.stringify({
      followed_tag_names: [
        {
          bg_color_hex: '#000000',
          id: 715,
          name: 'discuss',
          text_color_hex: '#ffffff',
        },
        {
          bg_color_hex: '#f7df1e',
          id: 6,
          name: 'javascript',
          text_color_hex: '#000000',
        },
      ],
    });
    document.body.setAttribute('data-user', dataUser);
    const meta = document.createElement('meta');
    meta.setAttribute('name', 'csrf-token');
    document.body.appendChild(meta);
    fetch.mockResponseOnce(fakeResponse);
    const context = await deep(<Onboarding />, { depth: 10 });
    context.setState({ pageNumber: 2 });
    context.rerender();
    console.log(context.find('.onboarding-tag-link-follow'));
  });
});
