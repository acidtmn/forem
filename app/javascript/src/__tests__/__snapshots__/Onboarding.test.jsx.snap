// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Onboarding /> Follow dev members page BACK button works 1`] = `
preact-render-spy (1 nodes)
-------
<div
  class="global-modal"
  style="display:none"
>
  <div class="global-modal-bg">
    <button
      class="close-button"
      onClick={[Function bound closeOnboarding]}
    >
      <img
        src=""
        alt="cancel button"
       />
    </button>
  </div>
  <div class="global-modal-inner">
    <div class="modal-header">
      <div class="triangle-isosceles">FOLLOW TAGS</div>
    </div>
    <div class="modal-body">
      <div
        id="sloan-mascot-onboarding-area"
        class="sloan-bar wiggle"
      >
        <img
          src="https://res.cloudinary.com/practicaldev/image/fetch/s--iiubRINO--/c_imagga_scale,f_auto,fl_progressive,q_auto,w_300/https://practicaldev-herokuapp-com.freetls.fastly.net/assets/sloan.png"
          class="sloan-img"
         />
      </div>
      <div class="body-message">
        <OnboardingFollowTags
          userData={
            Object {
              "followed_tag_names": Array [
                Object {
                  "bg_color_hex": "#000000",
                  "id": 715,
                  "name": "discuss",
                  "text_color_hex": "#ffffff"
                },
                Object {
                  "bg_color_hex": "#f7df1e",
                  "id": 6,
                  "name": "javascript",
                  "text_color_hex": "#000000"
                }
              ]
            }
          }
          allTags={
            Array [
              Object {
                "bg_color_hex": "#000000",
                "following": false,
                "id": 715,
                "name": "discuss",
                "text_color_hex": "#ffffff"
              },
              Object {
                "bg_color_hex": "#f7df1e",
                "following": false,
                "id": 6,
                "name": "javascript",
                "text_color_hex": "#000000"
              },
              Object {
                "bg_color_hex": "#2a2566",
                "following": false,
                "id": 630,
                "name": "career",
                "text_color_hex": "#ffffff"
              }
            ]
          }
          handleFollowTag={[Function bound handleFollowTag]}
        >
        </OnboardingFollowTags>
      </div>
    </div>
    <div class="modal-footer">
      <div class="modal-footer-left">
        <button
          class="button cta"
          onClick={[Function bound handleBackButton]}
        >
           BACK 
        </button>
      </div>
      <div class="modal-footer-center">
        <div class="pageindicators">
          <div class="pageindicator pageindicator--active"></div>
          <div class="pageindicator"></div>
          <div class="pageindicator"></div>
        </div>
      </div>
      <div class="modal-footer-right">
        <button
          class="button cta"
          onClick={[Function bound handleNextButton]}
          onMouseOver={[Function bound handleNextHover]}
          onFocus={[Function bound handleNextHover]}
        >
          NEXT
        </button>
      </div>
    </div>
  </div>
</div>

`;

exports[`<Onboarding /> Follow dev members page NEXT button works 1`] = `
preact-render-spy (1 nodes)
-------
<div
  class="global-modal"
  style="display:none"
>
  <div class="global-modal-bg">
    <button
      class="close-button"
      onClick={[Function bound closeOnboarding]}
    >
      <img
        src=""
        alt="cancel button"
       />
    </button>
  </div>
  <div class="global-modal-inner">
    <div class="modal-header">
      <div class="triangle-isosceles">CREATE YOUR PROFILE</div>
    </div>
    <div class="modal-body">
      <div
        id="sloan-mascot-onboarding-area"
        class="sloan-bar wiggle"
      >
        <img
          src="https://res.cloudinary.com/practicaldev/image/fetch/s--iiubRINO--/c_imagga_scale,f_auto,fl_progressive,q_auto,w_300/https://practicaldev-herokuapp-com.freetls.fastly.net/assets/sloan.png"
          class="sloan-img"
         />
      </div>
      <div class="body-message">
        <OnboardingProfile onChange={[Function bound handleProfileChange]}></OnboardingProfile>
      </div>
    </div>
    <div class="modal-footer">
      <div class="modal-footer-left">
        <button
          class="button cta"
          onClick={[Function bound handleBackButton]}
        >
           BACK 
        </button>
      </div>
      <div class="modal-footer-center">
        <div class="pageindicators">
          <div class="pageindicator"></div>
          <div class="pageindicator"></div>
          <div class="pageindicator pageindicator--active"></div>
        </div>
      </div>
      <div class="modal-footer-right">
        <button
          class="button cta"
          onClick={[Function bound handleNextButton]}
          onMouseOver={[Function bound handleNextHover]}
          onFocus={[Function bound handleNextHover]}
        >
          NEXT
        </button>
      </div>
    </div>
  </div>
</div>

`;

exports[`<Onboarding /> Follow dev members page renders properly 1`] = `
preact-render-spy (1 nodes)
-------
<div
  class="global-modal"
  style="display:none"
>
  <div class="global-modal-bg">
    <button
      class="close-button"
      onClick={[Function bound closeOnboarding]}
    >
      <img
        src=""
        alt="cancel button"
       />
    </button>
  </div>
  <div class="global-modal-inner">
    <div class="modal-header">
      <div class="triangle-isosceles">FOLLOW DEVS</div>
    </div>
    <div class="modal-body">
      <div
        id="sloan-mascot-onboarding-area"
        class="sloan-bar wiggle"
      >
        <img
          src="https://res.cloudinary.com/practicaldev/image/fetch/s--iiubRINO--/c_imagga_scale,f_auto,fl_progressive,q_auto,w_300/https://practicaldev-herokuapp-com.freetls.fastly.net/assets/sloan.png"
          class="sloan-img"
         />
      </div>
      <div class="body-message">
        <OnboardingFollowUsers
          users={Array []}
          checkedUsers={Array []}
          handleCheckUser={[Function bound handleCheckUser]}
          handleCheckAllUsers={[Function bound handleCheckAllUsers]}
        >
        </OnboardingFollowUsers>
      </div>
    </div>
    <div class="modal-footer">
      <div class="modal-footer-left">
        <button
          class="button cta"
          onClick={[Function bound handleBackButton]}
        >
           BACK 
        </button>
      </div>
      <div class="modal-footer-center">
        <div class="pageindicators">
          <div class="pageindicator"></div>
          <div class="pageindicator pageindicator--active"></div>
          <div class="pageindicator"></div>
        </div>
      </div>
      <div class="modal-footer-right">
        <button
          class="button cta"
          onClick={[Function bound handleNextButton]}
          onMouseOver={[Function bound handleNextHover]}
          onFocus={[Function bound handleNextHover]}
        >
          NEXT
        </button>
      </div>
    </div>
  </div>
</div>

`;

exports[`<Onboarding /> Follow tag page BACK button works 1`] = `
preact-render-spy (1 nodes)
-------
<div
  class="global-modal"
  style="display:none"
>
  <div class="global-modal-bg">
    <button
      class="close-button"
      onClick={[Function bound closeOnboarding]}
    >
      <img
        src=""
        alt="cancel button"
       />
    </button>
  </div>
  <div class="global-modal-inner">
    <div class="modal-header">
      <div class="triangle-isosceles">WELCOME!</div>
    </div>
    <div class="modal-body">
      <div
        id="sloan-mascot-onboarding-area"
        class="sloan-bar wiggle"
      >
        <img
          src="https://res.cloudinary.com/practicaldev/image/fetch/s--iiubRINO--/c_imagga_scale,f_auto,fl_progressive,q_auto,w_300/https://practicaldev-herokuapp-com.freetls.fastly.net/assets/sloan.png"
          class="sloan-img"
         />
      </div>
      <div class="body-message">
        <div class="onboarding-initial-welcome">
          <p>Thank you for joining the DEV Community.</p>
          <p>
            Keep up with the people and software trends you care about. ❤️
          </p>
          <p>
            <em>
              <strong class="green">Let's get started!</strong>
            </em>
          </p>
        </div>
      </div>
    </div>
    <div class="modal-footer">
      <div class="modal-footer-left"></div>
      <div class="modal-footer-center">
        <div class="pageindicators">
          <div class="pageindicator"></div>
          <div class="pageindicator"></div>
          <div class="pageindicator"></div>
        </div>
      </div>
      <div class="modal-footer-right">
        <button
          class="button cta"
          onClick={[Function bound handleNextButton]}
          onMouseOver={[Function bound handleNextHover]}
          onFocus={[Function bound handleNextHover]}
        >
          NEXT
        </button>
      </div>
    </div>
  </div>
</div>

`;

exports[`<Onboarding /> Follow tag page NEXT button works 1`] = `
preact-render-spy (1 nodes)
-------
<div
  class="global-modal"
  style="display:none"
>
  <div class="global-modal-bg">
    <button
      class="close-button"
      onClick={[Function bound closeOnboarding]}
    >
      <img
        src=""
        alt="cancel button"
       />
    </button>
  </div>
  <div class="global-modal-inner">
    <div class="modal-header">
      <div class="triangle-isosceles">FOLLOW DEVS</div>
    </div>
    <div class="modal-body">
      <div
        id="sloan-mascot-onboarding-area"
        class="sloan-bar wiggle"
      >
        <img
          src="https://res.cloudinary.com/practicaldev/image/fetch/s--iiubRINO--/c_imagga_scale,f_auto,fl_progressive,q_auto,w_300/https://practicaldev-herokuapp-com.freetls.fastly.net/assets/sloan.png"
          class="sloan-img"
         />
      </div>
      <div class="body-message">
        <div class="onboarding-user-container">
          <div class="onboarding-user-cta">
            Here are some suggestions based on your interests
          </div>
          <div class="onboarding-user-list">
            <div class="onboarding-user-list-header">
              <div class="onboarding-user-list-key">Follow All</div>
              <div class="onboarding-user-list-checkbox">
                <button
                  id="onboarding-user-follow-all-btn"
                  onClick={[Function bound handleAllClick]}
                  class="checked"
                >
                  ✓
                </button>
              </div>
            </div>
            <div class="onboarding-user-list-body">
              <div class="onboarding-user-list-row">
                <div class="onboarding-user-list-key">
                  <img
                    src="ben.jpg"
                    alt="Ben Halpern"
                   />
                  <div>Ben Halpern</div>
                  <div class="onboarding-user-list-row__summary">
                    <em></em>
                  </div>
                </div>
                <div class="onboarding-user-list-checkbox">
                  <button
                    onClick={[Function bound bound handleCheckUser]}
                    class="checked"
                  >
                    ✓
                  </button>
                </div>
              </div>
              <div class="onboarding-user-list-row">
                <div class="onboarding-user-list-key">
                  <img
                    src="clown.jpg"
                    alt="Krusty the Clown"
                   />
                  <div>Krusty the Clown</div>
                  <div class="onboarding-user-list-row__summary">
                    <em></em>
                  </div>
                </div>
                <div class="onboarding-user-list-checkbox">
                  <button
                    onClick={[Function bound bound handleCheckUser]}
                    class="checked"
                  >
                    ✓
                  </button>
                </div>
              </div>
              <div class="onboarding-user-list-row">
                <div class="onboarding-user-list-key">
                  <img
                    src="dev.jpg"
                    alt="dev.to staff"
                   />
                  <div>dev.to staff</div>
                  <div class="onboarding-user-list-row__summary">
                    <em></em>
                  </div>
                </div>
                <div class="onboarding-user-list-checkbox">
                  <button
                    onClick={[Function bound bound handleCheckUser]}
                    class="checked"
                  >
                    ✓
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div class="modal-footer">
      <div class="modal-footer-left">
        <button
          class="button cta"
          onClick={[Function bound handleBackButton]}
        >
           BACK 
        </button>
      </div>
      <div class="modal-footer-center">
        <div class="pageindicators">
          <div class="pageindicator"></div>
          <div class="pageindicator pageindicator--active"></div>
          <div class="pageindicator"></div>
        </div>
      </div>
      <div class="modal-footer-right">
        <button
          class="button cta"
          onClick={[Function bound handleNextButton]}
          onMouseOver={[Function bound handleNextHover]}
          onFocus={[Function bound handleNextHover]}
        >
          NEXT
        </button>
      </div>
    </div>
  </div>
</div>

`;

exports[`<Onboarding /> Follow tag page renders properly 1`] = `
preact-render-spy (1 nodes)
-------
<div
  class="global-modal"
  style="display:none"
>
  <div class="global-modal-bg">
    <button
      class="close-button"
      onClick={[Function bound closeOnboarding]}
    >
      <img
        src=""
        alt="cancel button"
       />
    </button>
  </div>
  <div class="global-modal-inner">
    <div class="modal-header">
      <div class="triangle-isosceles">FOLLOW TAGS</div>
    </div>
    <div class="modal-body">
      <div
        id="sloan-mascot-onboarding-area"
        class="sloan-bar wiggle"
      >
        <img
          src="https://res.cloudinary.com/practicaldev/image/fetch/s--iiubRINO--/c_imagga_scale,f_auto,fl_progressive,q_auto,w_300/https://practicaldev-herokuapp-com.freetls.fastly.net/assets/sloan.png"
          class="sloan-img"
         />
      </div>
      <div class="body-message">
        <div class="tags-slide">
          <div class="onboarding-user-cta">Personalize your home feed</div>
          <div class="tags-col-container">
            <div
              class="onboarding-tag-container"
              id="onboarding-tag-container-discuss"
              style="background: "
            >
              <a
                class="onboarding-tag-link"
                href="#"
                style="color:"
                onClick={[Function bound onClick]}
              >
                #discuss
              </a>
              <a
                class="onboarding-tag-link-follow"
                href="#"
                id="onboarding-tag-link-follow-discuss"
                style="color:"
                onClick={[Function bound onClick]}
              >
                +
              </a>
            </div>
            <div
              class="onboarding-tag-container"
              id="onboarding-tag-container-javascript"
              style="background: "
            >
              <a
                class="onboarding-tag-link"
                href="#"
                style="color:"
                onClick={[Function bound onClick]}
              >
                #javascript
              </a>
              <a
                class="onboarding-tag-link-follow"
                href="#"
                id="onboarding-tag-link-follow-javascript"
                style="color:"
                onClick={[Function bound onClick]}
              >
                +
              </a>
            </div>
            <div
              class="onboarding-tag-container"
              id="onboarding-tag-container-career"
              style="background: "
            >
              <a
                class="onboarding-tag-link"
                href="#"
                style="color:"
                onClick={[Function bound onClick]}
              >
                #career
              </a>
              <a
                class="onboarding-tag-link-follow"
                href="#"
                id="onboarding-tag-link-follow-career"
                style="color:"
                onClick={[Function bound onClick]}
              >
                +
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div class="modal-footer">
      <div class="modal-footer-left">
        <button
          class="button cta"
          onClick={[Function bound handleBackButton]}
        >
           BACK 
        </button>
      </div>
      <div class="modal-footer-center">
        <div class="pageindicators">
          <div class="pageindicator pageindicator--active"></div>
          <div class="pageindicator"></div>
          <div class="pageindicator"></div>
        </div>
      </div>
      <div class="modal-footer-right">
        <button
          class="button cta"
          onClick={[Function bound handleNextButton]}
          onMouseOver={[Function bound handleNextHover]}
          onFocus={[Function bound handleNextHover]}
        >
          NEXT
        </button>
      </div>
    </div>
  </div>
</div>

`;

exports[`<Onboarding /> Welcome page NEXT button should work 1`] = `
preact-render-spy (1 nodes)
-------
<div
  class="global-modal"
  style="display:none"
>
  <div class="global-modal-bg">
    <button
      class="close-button"
      onClick={[Function bound closeOnboarding]}
    >
      <img
        src=""
        alt="cancel button"
       />
    </button>
  </div>
  <div class="global-modal-inner">
    <div class="modal-header">
      <div class="triangle-isosceles">FOLLOW TAGS</div>
    </div>
    <div class="modal-body">
      <div
        id="sloan-mascot-onboarding-area"
        class="sloan-bar wiggle"
      >
        <img
          src="https://res.cloudinary.com/practicaldev/image/fetch/s--iiubRINO--/c_imagga_scale,f_auto,fl_progressive,q_auto,w_300/https://practicaldev-herokuapp-com.freetls.fastly.net/assets/sloan.png"
          class="sloan-img"
         />
      </div>
      <div class="body-message">
        <div class="tags-slide">
          <div class="onboarding-user-cta">Personalize your home feed</div>
          <div class="tags-col-container">
            <div
              class="onboarding-tag-container"
              id="onboarding-tag-container-discuss"
              style="background: "
            >
              <a
                class="onboarding-tag-link"
                href="#"
                style="color:"
                onClick={[Function bound onClick]}
              >
                #discuss
              </a>
              <a
                class="onboarding-tag-link-follow"
                href="#"
                id="onboarding-tag-link-follow-discuss"
                style="color:"
                onClick={[Function bound onClick]}
              >
                +
              </a>
            </div>
            <div
              class="onboarding-tag-container"
              id="onboarding-tag-container-javascript"
              style="background: "
            >
              <a
                class="onboarding-tag-link"
                href="#"
                style="color:"
                onClick={[Function bound onClick]}
              >
                #javascript
              </a>
              <a
                class="onboarding-tag-link-follow"
                href="#"
                id="onboarding-tag-link-follow-javascript"
                style="color:"
                onClick={[Function bound onClick]}
              >
                +
              </a>
            </div>
            <div
              class="onboarding-tag-container"
              id="onboarding-tag-container-career"
              style="background: "
            >
              <a
                class="onboarding-tag-link"
                href="#"
                style="color:"
                onClick={[Function bound onClick]}
              >
                #career
              </a>
              <a
                class="onboarding-tag-link-follow"
                href="#"
                id="onboarding-tag-link-follow-career"
                style="color:"
                onClick={[Function bound onClick]}
              >
                +
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div class="modal-footer">
      <div class="modal-footer-left">
        <button
          class="button cta"
          onClick={[Function bound handleBackButton]}
        >
           BACK 
        </button>
      </div>
      <div class="modal-footer-center">
        <div class="pageindicators">
          <div class="pageindicator pageindicator--active"></div>
          <div class="pageindicator"></div>
          <div class="pageindicator"></div>
        </div>
      </div>
      <div class="modal-footer-right">
        <button
          class="button cta"
          onClick={[Function bound handleNextButton]}
          onMouseOver={[Function bound handleNextHover]}
          onFocus={[Function bound handleNextHover]}
        >
          NEXT
        </button>
      </div>
    </div>
  </div>
</div>

`;

exports[`<Onboarding /> Welcome page renders properly 1`] = `
preact-render-spy (1 nodes)
-------
<div
  class="global-modal"
  style="display:none"
>
  <div class="global-modal-bg">
    <button
      class="close-button"
      onClick={[Function bound closeOnboarding]}
    >
      <img
        src=""
        alt="cancel button"
       />
    </button>
  </div>
  <div class="global-modal-inner">
    <div class="modal-header">
      <div class="triangle-isosceles">WELCOME!</div>
    </div>
    <div class="modal-body">
      <div
        id="sloan-mascot-onboarding-area"
        class="sloan-bar wiggle"
      >
        <img
          src="https://res.cloudinary.com/practicaldev/image/fetch/s--iiubRINO--/c_imagga_scale,f_auto,fl_progressive,q_auto,w_300/https://practicaldev-herokuapp-com.freetls.fastly.net/assets/sloan.png"
          class="sloan-img"
         />
      </div>
      <div class="body-message">
        <div class="onboarding-initial-welcome">
          <p>Thank you for joining the DEV Community.</p>
          <p>
            Keep up with the people and software trends you care about. ❤️
          </p>
          <p>
            <em>
              <strong class="green">Let's get started!</strong>
            </em>
          </p>
        </div>
      </div>
    </div>
    <div class="modal-footer">
      <div class="modal-footer-left"></div>
      <div class="modal-footer-center">
        <div class="pageindicators">
          <div class="pageindicator"></div>
          <div class="pageindicator"></div>
          <div class="pageindicator"></div>
        </div>
      </div>
      <div class="modal-footer-right">
        <button
          class="button cta"
          onClick={[Function bound handleNextButton]}
          onMouseOver={[Function bound handleNextHover]}
          onFocus={[Function bound handleNextHover]}
        >
          NEXT
        </button>
      </div>
    </div>
  </div>
</div>

`;

exports[`<Onboarding /> allows user to interact properly 1`] = `
FindWrapper {
  "0": VNode {
    "attributes": Object {
      "className": "close-button",
      "onClick": [Function],
    },
    "children": Array [
      VNode {
        "attributes": Object {
          "alt": "cancel button",
          "src": "",
        },
        "children": Array [],
        "key": undefined,
        "nodeName": "img",
      },
    ],
    "key": undefined,
    "nodeName": "button",
  },
  "1": VNode {
    "attributes": Object {
      "className": "button cta",
      "onClick": [Function],
    },
    "children": Array [
      " BACK ",
    ],
    "key": undefined,
    "nodeName": "button",
  },
  "2": VNode {
    "attributes": Object {
      "className": "button cta",
      "onClick": [Function],
      "onFocus": [Function],
      "onMouseOver": [Function],
    },
    "children": Array [
      "NEXT",
    ],
    "key": undefined,
    "nodeName": "button",
  },
  "length": 3,
}
`;

exports[`<Onboarding /> display if there is current user 1`] = `
preact-render-spy (1 nodes)
-------
<div
  class="global-modal"
  style="display:none"
>
  <div class="global-modal-bg">
    <button
      class="close-button"
      onClick={[Function bound closeOnboarding]}
    >
      <img
        src=""
        alt="cancel button"
       />
    </button>
  </div>
  <div class="global-modal-inner">
    <div class="modal-header">
      <div class="triangle-isosceles">WELCOME!</div>
    </div>
    <div class="modal-body">
      <div
        id="sloan-mascot-onboarding-area"
        class="sloan-bar wiggle"
      >
        <img
          src="https://res.cloudinary.com/practicaldev/image/fetch/s--iiubRINO--/c_imagga_scale,f_auto,fl_progressive,q_auto,w_300/https://practicaldev-herokuapp-com.freetls.fastly.net/assets/sloan.png"
          class="sloan-img"
         />
      </div>
      <div class="body-message">
        <OnboardingWelcome></OnboardingWelcome>
      </div>
    </div>
    <div class="modal-footer">
      <div class="modal-footer-left"></div>
      <div class="modal-footer-center">
        <div class="pageindicators">
          <div class="pageindicator"></div>
          <div class="pageindicator"></div>
          <div class="pageindicator"></div>
        </div>
      </div>
      <div class="modal-footer-right">
        <button
          class="button cta"
          onClick={[Function bound handleNextButton]}
          onMouseOver={[Function bound handleNextHover]}
          onFocus={[Function bound handleNextHover]}
        >
          NEXT
        </button>
      </div>
    </div>
  </div>
</div>

`;

exports[`<Onboarding /> when user is not logged in nothing should be rendered 1`] = `
preact-render-spy (1 nodes)
-------
{undefined}

`;
